server/api/src/rsDataObjTruncate.cpp

```diff
@@ -24,150 +24,161 @@
 
 #include "irods/irods_resource_backport.hpp"
 
+#pragma clang diagnostic push
+#pragma clang diagnostic ignored "-Wdeprecated-declarations"
+
+namespace
+{
+    int l3Truncate(rsComm_t* rsComm, dataObjInp_t* dataObjTruncateInp, dataObjInfo_t* dataObjInfo)
+    {
+        fileOpenInp_t fileTruncateInp;
+        int status;
+
+        // =-=-=-=-=-=-=-
+        // extract the host location from the resource hierarchy
+        std::string location;
+        irods::error ret = irods::get_loc_for_hier_string(dataObjInfo->rescHier, location);
+        if (!ret.ok()) {
+            irods::log(PASSMSG("l3Truncate - failed in get_loc_for_hier_string", ret));
+            return -1;
+        }
 
-
-int
-rsDataObjTruncate( rsComm_t *rsComm, dataObjInp_t *dataObjTruncateInp ) {
-    int status;
-    dataObjInfo_t *dataObjInfoHead = NULL;
-    int remoteFlag;
-    rodsServerHost_t *rodsServerHost;
-
-    remoteFlag = getAndConnRemoteZone( rsComm, dataObjTruncateInp,
-                                       &rodsServerHost, REMOTE_OPEN );
-
-    if ( remoteFlag < 0 ) {
-        return remoteFlag;
-    }
-    else if ( remoteFlag == REMOTE_HOST ) {
-        status = rcDataObjTruncate( rodsServerHost->conn, dataObjTruncateInp );
-        return status;
-    }
-
-    dataObjTruncateInp->openFlags = O_WRONLY;  /* set the permission checking */
-    status = getDataObjInfoIncSpecColl( rsComm, dataObjTruncateInp,
-                                        &dataObjInfoHead );
-
-    if ( status < 0 ) {
+        if (getStructFileType(dataObjInfo->specColl) >= 0) {
+            subFile_t subFile;
+            memset(&subFile, 0, sizeof(subFile));
```

Comment:
Do we prefer `std::memset(...)`?

Reply:
The only reason this appears as a new line is because I moved the function into an anonymous namespace and ran clang-format on the file. I don't particularly want to update anything internally in this API other than the addition of the deprecation message to the rError stack.

That being said, I will update it if you feel strongly about it.

Reply:
Gotcha. Think I'd lean toward throwing a `std` on there.

Reply:
Well, actually, we could just zero-initialize the struct and avoid calling `(std::)memset` entirely here, like so:
```c++
subFile_t subFile{};
```
Thoughts?

Reply:
Even better.

Reply:
Done. Reduced some scopes too